server/export.go

```diff
@@ -0,0 +1,133 @@
+package server
+
+// DONTCOVER
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"strconv"
+
+	rdk_genutiltypes "github.com/dymensionxyz/dymension-rdk/x/genutil/types"
+	"github.com/spf13/cobra"
+	tmproto "github.com/tendermint/tendermint/proto/tendermint/types"
+
+	"github.com/cosmos/cosmos-sdk/client/flags"
+	"github.com/cosmos/cosmos-sdk/server"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/dymensionxyz/dymension-rdk/server/types"
+)
+
+const (
+	FlagHeight           = "height"
+	FlagForZeroHeight    = "for-zero-height"
+	FlagJailAllowedAddrs = "jail-allowed-addrs"
+	flagTraceStore       = "trace-store"
+)
+
+// ExportCmd dumps app state to JSON.
+func ExportCmd(appExporter types.AppExporter, defaultNodeHome string) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "export",
+		Short: "Export state to JSON",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			serverCtx := server.GetServerContextFromCmd(cmd)
+			config := serverCtx.Config
+
+			homeDir, _ := cmd.Flags().GetString(flags.FlagHome)
+			config.SetRoot(homeDir)
+
+			if _, err := os.Stat(config.GenesisFile()); os.IsNotExist(err) {
+				return err
+			}
+
+			db, err := openDB(config.RootDir, server.GetAppDBBackend(serverCtx.Viper))
+			if err != nil {
+				return err
+			}
+
+			if appExporter == nil {
+				if _, err := fmt.Fprintln(os.Stderr, "WARNING: App exporter not defined. Returning genesis file."); err != nil {
+					return err
+				}
+
+				genesis, err := os.ReadFile(config.GenesisFile())
+				if err != nil {
+					return err
+				}
+
+				fmt.Println(string(genesis))
+				return nil
+			}
+
+			genDoc, err := rdk_genutiltypes.GenesisDocFromFile(config.GenesisFile())
+			if err != nil {
+				fmt.Println("Failed to read genesis doc from file", err)
```

Comment:
why sometimes stderr, sometimes stdout?

Reply:
it's just a function taken from cosmos-sdk with a little change